packages/react-reconciler/src/ReactFiberCacheComponent.new.js

```diff
@@ -91,22 +161,32 @@ export function pushRootCachePool(root: FiberRoot) {
   // from `root.pooledCache`. If it's currently `null`, we will lazily
   // initialize it the first type it's requested. However, we only mutate
   // the root itself during the complete/unwind phase of the HostRoot.
-  pooledCache = root.pooledCache;
+  const rootCache = root.pooledCache;
+  if (rootCache != null) {
+    trace('pushRootCachePool');
+    pooledCache = rootCache;
+    root.pooledCache = null;
```

Comment:
for symmetry with `popRootCachePool()`, we unset `root.pooledCache` here

Reply:
Fine either way, it should always be `null` here because we always `pop` when we exit during unwind or complete. In the context implementation, what we'll do is warn in dev if the stack isn't empty at the end, but not in prod. That way we don't have to run extra checks but if we get a signal if we mess up in dev.

Reply:
^ my comment was about line 170, I misread which line you were commenting on